24_clustering.Rmd

```diff
@@ -221,97 +196,122 @@ tse <- GlobalPatterns
 
-dmm_group
-```
+```{r dmm8}
+library(dplyr)
 
-Mixture weights (rough measure of the cluster size).
+clusters <- round(metadata(tse_dmm)$DMM$prob, 1)
+clusters <- as.data.frame(cbind(clusters, levels(colData(tse_dmm)$SampleType)[colData(tse_dmm)$SampleType])) # add sample type information
+colnames(clusters) <- c("Group1", "Group2", "SampleType")
 
-```{r dmm7}
-DirichletMultinomial::mixturewt(bestFit)
+clusters %>%
+  group_by(SampleType) %>%
+  arrange(Group1) %>%
+  filter(row_number()==1)
 ```
 
-It's also possible to get the samples-cluster assignment probabilities: how
-probable it is that each sample belongs to each cluster
+We can also plot the driver Phyla in each group. In this case, it reflects the differences between environmental and human samples. 
 
-```{r dmm8}
-prob <- metadata(tse)$DMM$prob
-head(prob)
+```{r dmm9, fig.width=8, fig.height=4}
+# Get the estimates on how much each phyla contributes on each cluster
+best_model <- metadata(tse_dmm)$DMM$dmm[2]
+drivers <- as.data.frame(best_model[[1]]@fit$Estimate)
+
+drivers$phyla <- gsub("Phylum:", "", rownames(drivers)) # Clean phylum names
+
+plots <- c()
+for (i in 1:2) {
+  drivers <- drivers[order(drivers[[i]], decreasing = TRUE),]
+  p <- ggplot(head(drivers, 10), aes(x = reorder(head(phyla, 10), + head(drivers[[i]], 10)), y = head(drivers[[i]], 10))) +
+          geom_bar(stat = "identity", fill = "deeppink4", alpha = 0.5) +
+          coord_flip() + labs(title = paste("Top phyla in a group", i)) +
+          theme_light(base_size = 15) + xlab("") + ylab("") + ylim(0, 7)
+  print(p)
+}
 ```
 
-We can also know the contribution of each taxa to each component
 
-```{r dmm9}
-head(DirichletMultinomial::fitted(bestFit))
+We use `calculateDMNgroup` function to have an overview of the best model. The function groups samples by sample type and returns DMNGroup object that contains a summary.
+
+```{r dmm_group}
+dmm_group <- calculateDMNgroup(tse_dmm, variable = "SampleType", 
+                               assay.type = "counts", k = 2, 
+                               seed = .Machine$integer.max)
+
+dmm_group
 ```
 
-Finally, to be able to visualize our data and clusters, we start by 
-computing the euclidean PCoA and storing it as a data frame.
+Mixture weights can be used for having a rough approximation of the cluster size.
 
 ```{r dmm10}
-# Do clr transformation. Pseudocount is added, because data contains zeros.
+DirichletMultinomial::mixturewt(bestFit)
+```
+
+### PCoA with DMM clusters
+
+In this section we show how to calculate principal coordinates for clr transformed abundance data. We use Aitchison distance as a distance metrics that calculates Euclidean distances for clr transformed compositions. In the visualization section, the samples are coloured based on their DMM clusters to demonstrate the distances between samples included in each DMM cluster.
```

Comment:
I may misunderstand something but DMM assumes count data and it has its own model for clustering. 

It does not use CLR data or Euclidean / Aitchison distances. It should be made clear what transformations & distances refer to PCoA, and which to DMM (and that these are distinct)

Reply:
Thanks for the comment, I fixed the text to be more clear. I think also using alternative experiments clarify the distinction in the code.